### ISSUE TITLE
seat-groups unassigns roles that it has no rules for

### ISSUE THREAD

**Issue by dougestey** [herpaderpaldent/seat-groups#40]
seat-groups revokes roles that it does not have a "rule" (group) for.

Repro:
- Install seat-groups
- Create a role with permission "superuser"
- Add a user/group ID to that role
- Wait for seat-group job to run
- User is unassigned from the role

This seems heavy-handed, because it makes granular role permission outside of the seat-groups UI unusable. Is this by design?

(In my case, I am managing certain roles via Seat's API through a Discord bot. I would love to be able to manage seat-groups programmatically, but this package does not seem to expose API endpoints to do so.)

**Comment by herpaderpaldent** [comment]
Hey @dougestey 
Thank you for contacting me. Feedback like yours really help improving this plugin. 

>Is this by design?
Yes and no. 
Yes as it replaces the vanilla functionality of assigning user_groups to roles. No as SeAT-Groups escapes Users that have the Superuser-Role: https://github.com/herpaderpaldent/seat-groups/blob/master/src/Jobs/GroupSync.php#L81
Note the roles name is `superuser` not the permission. 

> but this package does not seem to expose API endpoints to do so
Yes at the moment there is no API for creating or maintaining seat-groups. However this package is scheduled to become part of SeAT 3.1 at which point i'm sure there will be an API. Maybe @warlof can give me a hand before that to expose functionality to SeAT API. I am not very experienced with it but he is. 
Meanwhile you might want your users to use the seat-groups UI to optin or apply to a role/group. And let your managers accept deny those applications. Additionally to this you can use hidden groups for manual matching.

**Comment by dougestey** [comment]
Thank you for the response @herpaderpaldent, as well as for your excellent work on this and the Discourse plugin - I've just started using both, but I'm finding them very useful.

I may actually end up forking this package and patching it with the role groups that I don't want managed by seat-groups in order to meet my needs for now. I noticed that recently you eliminated the need for publishing changes to the module source... if I were to patch the source, how can I trigger an update to the code used in production? Will a simple `supervisor restart` suffice?

Sorry to drift off topic here. Looking forward to SeAT 3.1.
